### DISCUSSION TITLE
LSP document highlight - multiple cursors

### DISCUSSION THREAD

**Discussion by ROMUSKING** [helix-editor/helix#4133] [General]
First time reporting issue, hope this is relevant. 

Moving after using Space-h highlight moves all cursors. Actions and input are replicated under each cursor

![Screenshot 2022-10-06 231409](https://user-images.githubusercontent.com/38523153/194428402-829a654e-d88f-49d7-ba05-95d9a3ce0016.png)
![Screenshot 2022-10-06 231344](https://user-images.githubusercontent.com/38523153/194428404-4bd70ac3-564b-4225-aec5-f1377b380608.png)
![Screenshot 2022-10-06 232038](https://user-images.githubusercontent.com/38523153/194429172-fbb004fb-abf7-4d78-9c13-957caa06e03b.png)
 

windows
helix 22.08.1 (98dd9c4f)
installed through cargo

**Comment by kirawi** [comment]
This is how multiple cursors work. What behaviour are you looking for?

**Reply by ROMUSKING** [reply]
i was expecting  Select -> do something -> back to normal business when moving (with the primary cursor), maybe with ESC, as this would change visual mode to normal?

**Reply by kirawi** [reply]
You can customize your bindings to automatically remove extra cursors on movement.

**Reply by ROMUSKING** [reply]
will look into that. thanks for the response and an awesome editor 👌
